### ISSUE TITLE
ingress doesn't quite work with AWS ALB

### ISSUE THREAD

**Issue by daenney** [ansible/tower-operator#230]
I deployed AWX using the operator without an ingress at first and could succesfully access the web UI etc using `kubectl port-forward`. Having switched to an ingress resource using an AWS ALB I am greeted with a blank page in my browser because it fails to fetch the CSS and JS bundles from `/static`.

My AWX resource looks like:
```yaml
spec:
  tower_admin_user: admin
  tower_create_preload_data: true
  tower_garbage_collect_secrets: false
  tower_hostname: <REDACTED>
  tower_image_pull_policy: IfNotPresent
  tower_ingress_annotations: |
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS":443}]'
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    kubernetes.io/ingress.class: alb
  tower_ingress_type: Ingress
  tower_loadbalancer_port: 80
  tower_loadbalancer_protocol: http
  tower_postgres_configuration_secret: awx-postgres-configuration
  tower_projects_persistence: false
  tower_projects_storage_access_mode: ReadWriteMany
  tower_projects_storage_size: 8Gi
  tower_replicas: 1
  tower_route_tls_termination_mechanism: Edge
  tower_task_privileged: false
```

This results in an ingress of:

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS":443}]'
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    kubectl.kubernetes.io/last-applied-configuration: '{"apiVersion":"extensions/v1beta1","kind":"Ingress","metadata":{"annotations":{"alb.ingress.kubernetes.io/listen-ports":"[{\"HTTPS\":443}]","alb.ingress.kubernetes.io/scheme":"internet-facing","alb.ingress.kubernetes.io/target-type":"ip","kubernetes.io/ingress.class":"alb"},"labels":{"app.kubernetes.io/component":"awx","app.kubernetes.io/managed-by":"awx-operator","app.kubernetes.io/name":"awx","app.kubernetes.io/part-of":"awx"},"name":"awx-ingress","namespace":"awx"},"spec":{"rules":[{"host":"<REDACTED>","http":{"paths":[{"backend":{"serviceName":"awx-service","servicePort":80},"path":"/"}]}}]}}'
    kubernetes.io/ingress.class: alb
  creationTimestamp: "2021-04-21T11:51:31Z"
  finalizers:
  - ingress.k8s.aws/resources
  generation: 1
  labels:
    app.kubernetes.io/component: awx
    app.kubernetes.io/managed-by: awx-operator
    app.kubernetes.io/name: awx
    app.kubernetes.io/part-of: awx
  name: awx-ingress
  namespace: awx
  ownerReferences:
  - apiVersion: awx.ansible.com/v1beta1
    kind: AWX
    name: awx
    uid: 1489cf00-d4c3-4f3e-9a1f-789e649c428c
  resourceVersion: "6048870"
  selfLink: /apis/extensions/v1beta1/namespaces/awx/ingresses/awx-ingress
  uid: 63ca8a03-2e64-4021-9549-9c5a0854ff11
spec:
  rules:
  - host: <REDACTED>
    http:
      paths:
      - backend:
          serviceName: awx-service
          servicePort: 80
        path: /
        pathType: ImplementationSpecific
status:
  loadBalancer:
    ingress:
    - hostname: <REDACTED>.elb.amazonaws.com
```

The rules this generates for an ALB in AWS are:

```
1 IF
Path is/
Host <REDACTED>
THEN
Forward to
k8s-awx-awxservi-7d462a1e8e: 1 (100%)
Group-level stickiness: Off

last HTTPS 443: default action
This rule cannot be moved or deleted
IF
Requests otherwise not routed
THEN
Return fixed response 404 
```

Looking at my browser console:

```
GET <REDACTED>/static/css/2.4f9d8bcb.chunk.css
[HTTP/2 404 Not Found 23ms]

GET <REDACTED>/static/js/runtime-main.a1234c1a.js
[HTTP/2 404 Not Found 23ms]

GET <REDACTED>/static/js/2.89b6cada.chunk.js
[HTTP/2 404 Not Found 23ms]

GET <REDACTED>/static/js/main.2594bb50.chunk.js
```

It seems it's not able to fetch the static resources, which I would expect are served by the container.

Updating the rule in AWS to use `/*` for the path, instead of just `/` works. The ALB ingress controller does not support a type of `pathType: Prefix`, and it doesn't seem like I would be able to control this using the awx-operator anyway.

I would like to send a patch to resolve it, and in the case of the AWS ALB ingress controller what we'd need is to set the `path` of the ingress resource to `/*` instead. This should probably only done if the ingress is being managed by the AWS ALB for now? How could we check for that aside from just looking for "alb" in ` tower_ingress_annotations`?

**Comment by daenney** [comment]
An alternative could be something like `tower_ingress_path_wildcard: yes|no`?

**Comment by ksquires** [comment]
i first came across experiencing this issue this week (which has been causing me a lot of grief).  seeing how this has been opened for a month and half, I'm sort of sad.  But @daenney , maybe tag your issue as a bug? maybe it will get better visibility

thank you for the quick and simple workaround, too!

**Comment by daenney** [comment]
@ksquires I'm a "contributor", not a member of this org/repo so unfortunately I'm not able to set labels. A maintainer/member will have to do that.

**Comment by daenney** [comment]
This got fixed in #464.
